### PR TITLE
feat: support dependency list for decorator hybrid jobs

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

support listing dependencies by name instead of in a requirements file for decorator jobs

*Testing done:*

unit and manual testing

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ajberdy** [comment]
Getting formatting errors unrelated to changes made here. Will fix and see what's going on with the description workflow too

File: src/braket/jobs/hybrid_job.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def hybrid_job(
             modules to be included. Any references to members of these modules in the hybrid job
             algorithm code will be serialized as part of the algorithm code. Default value `[]`
 
-        dependencies (str | Path): Path (absolute or relative) to a requirements.txt
+        dependencies (str | Path | list[str]): Path (absolute or relative) to a requirements.txt
             file to be used for the hybrid job.
```
[the remark this answers is not in the document]
Comment (kshitijc) [reply src/braket/jobs/hybrid_job.py:89]: Could we also add some detail on what happens if there is already a `requirements.txt` file that is being included through the `include_modules` parameter and you additionally specify dependencies as a list of strings?

[the remark this answers is not in the document]
Comment (ajberdy) [reply src/braket/jobs/hybrid_job.py:89]: `include_modules` only allows python modules (`my_python_package.sub_module` etc) and does not directly access the file system, so there is no way for users to provide a `requirements.txt` through include modules. `include_modules` is a code-only construct

[the remark this answers is not in the document]
Comment (kshitijc) [reply src/braket/jobs/hybrid_job.py:89]: Ah, yes. I forgot about that. Happy to approve once the docstring is updated.

**Review by kshitijc — APPROVED** [review]

**Review by kshitijc — APPROVED** [review]

**Pull request merged**